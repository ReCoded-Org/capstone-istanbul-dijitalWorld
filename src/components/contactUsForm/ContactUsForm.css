.contactUsContainer {
  margin: 5% 5%;
}

.contactUsHero {
  color: #f67280;
}
.contactUsButton {
  background-color: #6c567b;
  color: white;
  cursor: pointer;
  width: 10rem;
  height: 2.5rem;
  font-size: 1.5rem;
  border-radius: 3.125rem;
  border-style: hidden;
  box-shadow: 0 0.6875rem 1.5rem -0.4375rem #b3b3b3;
}

.contactUsButton:hover {
  background-color: #f8b195;
}
