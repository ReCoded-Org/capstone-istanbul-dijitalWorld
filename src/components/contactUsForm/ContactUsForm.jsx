import React, { useState } from 'react';
import './ContactUsForm.css';
import database from '../../firebase';
import { Form, Col, Row, Alert } from 'react-bootstrap';
import { useTranslation } from 'react-i18next';

export default function ContactUs() {
  const { t } = useTranslation();
  const [email, setEmail] = useState();
  const [message, setMessage] = useState();
  const [show, setShow] = useState(false);

  function ThankYouPopUp() {
    return (
      <>
        <Alert show={show} variant="success">
          <Alert.Heading>Thank You!</Alert.Heading>
          <p>
            Thank you for submitting this message.
            <br />
            Our admins will review your message and get back to you if needed.
          </p>
          <hr />
        </Alert>
      </>
    );
  }

  const handleEmailInput = async (e) => {
    setEmail(e.target.value);
  };

  const handleMessageInput = async (e) => {
    setMessage(e.target.value);
  };

  const handleSubmitMessage = (e) => {
    e.preventDefault();

    const newMessage = database.collection('contactUs').doc();
    newMessage.set({
      email: email,
      message: message,
    });
    setShow(true);
  };

  return (
    <Form onSubmit={handleSubmitMessage}>
      <Row>
        <Col>
          <h3 className="contactUsHero">{t('contactUs.getInTouch.0')}</h3>
          <h5>{t('contactUs.getInTouch.1')}</h5>

          <Form.Group>
            <Form.Control
              size="lg"
              type="email"
              placeholder={t('contactUs.email')}
              onChange={handleEmailInput}
              required
            />
          </Form.Group>
          <Form.Group>
            <Form.Control
              as="textarea"
              rows="10"
              placeholder={t('contactUs.message')}
              onChange={handleMessageInput}
              required
            />
          </Form.Group>
          <button type="submit" className="contactUsButton">
            {t('contactUs.button')}
          </button>
        </Col>
      </Row>
      <Row>
        <ThankYouPopUp />
      </Row>
    </Form>
  );
}
