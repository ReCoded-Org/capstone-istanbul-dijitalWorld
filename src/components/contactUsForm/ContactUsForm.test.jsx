import React from 'react';
import renderer from 'react-test-renderer';
import ContactUsForm from './ContactUsForm';

it('renders correctly', () => {
  const tree = renderer.create(<ContactUsForm />).toJSON();
  expect(tree).toMatchSnapshot();
});
