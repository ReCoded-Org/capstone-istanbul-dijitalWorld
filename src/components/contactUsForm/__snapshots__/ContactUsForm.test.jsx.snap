// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <h3
        className="contactUsHero"
      >
        contactUs.getInTouch.0
      </h3>
      <h5>
        contactUs.getInTouch.1
      </h5>
      <div
        className="form-group"
      >
        <input
          className="form-control form-control-lg"
          onChange={[Function]}
          placeholder="contactUs.email"
          required={true}
          type="email"
        />
      </div>
      <div
        className="form-group"
      >
        <textarea
          className="form-control"
          onChange={[Function]}
          placeholder="contactUs.message"
          required={true}
          rows="10"
        />
      </div>
      <button
        className="contactUsButton"
        type="submit"
      >
        contactUs.button
      </button>
    </div>
  </div>
  <div
    className="row"
  />
</form>
`;
